include/psa/crypto_config.h

```diff
@@ -429,11 +429,12 @@
  * \def MBEDTLS_THREADING_C
  *
  * Enable the threading abstraction layer.
- * By default Mbed TLS assumes it is used in a non-threaded environment or that
- * contexts are not shared between threads. If you do intend to use contexts
- * between threads, you will need to enable this layer to prevent race
- * conditions. See also our Knowledge Base article about threading:
- * https://mbed-tls.readthedocs.io/en/latest/kb/development/thread-safety-and-multi-threading
+ *
+ * \note You must enable this option if TF-PSA-Crypto runs in a
+ * multithreaded environment. Otherwise the PSA cryptography subsystem is
+ * not thread-safe. As an exception, this option can be disabled if all
+ * PSA crypto functions are ever called from a single thread. Note that
+ * this includes indirect calls, for example through PK.
```

Comment:
I am assuming that is the only thing required to enable for thread safety when using TF-PSA_Crrypto with latest MbedTLS? 

There is a a significant simplicaition of the documentation from #[10004](https://github.com/Mbed-TLS/mbedtls/pull/10004) which I assume is no longer applicable?

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/10004 has a lot of complexity to basically describe what you can do without PSA. Here, everything is built on PSA so threading is needed for everything.